src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  Post,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiQuery, ApiSecurity, ApiTags } from '@nestjs/swagger';
+import { AuthService } from './auth.service';
+import { AuthDTO } from './dto/auth.dto';
+import { TokenGuard } from './token.strategy';
+
+@ApiTags('Auth')
+@Controller('auth')
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get()
+  @UseGuards(TokenGuard)
+  @ApiSecurity('token')
+  @ApiQuery({ name: 'keyspace', required: true })
+  getTokens(@Query('keyspace') keyspace: string) {
+    return this.authService.getClientIds(keyspace);
+  }
+
+  @Post()
+  @UseGuards(TokenGuard)
+  @ApiSecurity('token')
+  register(@Body() body: AuthDTO) {
+    return this.authService.register(body);
+  }
+
+  @Delete()
+  @UseGuards(TokenGuard)
+  @ApiSecurity('token')
+  @ApiQuery({
+    name: 'token',
+    description: 'Token to delete',
+    required: true,
+  })
+  @HttpCode(204)
+  deleteClient(@Query() query) {
```

Comment:
I think this should be part of the path, I can change it in my PR. What do you think?